CvGameCoreDLL_Expansion2/CvMinorCivAI.cpp

```diff
@@ -5696,9 +5696,18 @@ int CvMinorCivAI::GetNumThreateningBarbarians()
 	int iCount = 0;
 
 	int iLoop = 0;
-	for(CvUnit* pLoopUnit = GET_PLAYER(BARBARIAN_PLAYER).firstUnit(&iLoop); NULL != pLoopUnit; pLoopUnit = GET_PLAYER(BARBARIAN_PLAYER).nextUnit(&iLoop))
-		if(pLoopUnit->plot()->isAdjacentTeam(GetPlayer()->getTeam()))
+	for (CvUnit* pLoopUnit = GET_PLAYER(BARBARIAN_PLAYER).firstUnit(&iLoop); NULL != pLoopUnit; pLoopUnit = GET_PLAYER(BARBARIAN_PLAYER).nextUnit(&iLoop))
+	{
+		if (!pLoopUnit)
```

Comment:
This already can't be NULL.

Reply:
yeah I suppose one of the null checks is unnecessary, but also there shouldn't be any real performance loss I think

Reply:
the crash was because one barbarian unit was in a non-existent tile

Reply:
But how? We probably should find the root cause.